### DISCUSSION TITLE
Is the ETH/USD Price Feed contract operational?

### DISCUSSION THREAD

**Discussion by PranavM191** [Cyfrin/foundry-full-course-cu#2674] [Q&A]
I am facing deployment issues using the price feed contract, I am currently on the Solidity Interfaces lesson where he tries to show us the price fetching using the ETH/USD price feed contract. I checked for the contract's operatability on etherscan, most of its recent transactions didnt go through. Can someone help?
![PriceFeedContract](https://github.com/user-attachments/assets/56561c92-611f-4a4c-96d7-bf9abc9c1b6f)

**Comment by EngrPips** [comment]
Please show your contract and the error you are facing.

**Reply by PranavM191** [reply]
//get funds from users
//withdraw users
//set a minimum value in USD

//SPDX-License-Identifier: MIT

pragma solidity ^0.8.24;

interface AggregatorV3Interface {
  function decimals() external view returns (uint8);

  function description() external view returns (string memory);

  function version() external view returns (uint256);

  function getRoundData(
    uint80 _roundId
  ) external view returns (uint80 roundId, int256 answer, uint256 startedAt, uint256 updatedAt, uint80 answeredInRound);

  function latestRoundData()
    external
    view
    returns (uint80 roundId, int256 answer, uint256 startedAt, uint256 updatedAt, uint80 answeredInRound);
}

contract FundMe{

    uint256 public minimumUSD = 5;

    function fund() public payable {

        //Allow users to send money
        //Have a minimum amount of $5
        //How do we send ETH to this
        require(msg.value>=minimumUSD, "didnt send enough ETH"); //1e18 = 1ETH = 1*10 ** 18

        // what is a revert ? 
        // It undoes any actions that have been done, and sends the remaining gas back
    }
    
    function getPrice() public {
        // with a contract we need its
        //Address 0x694AA1769357215DE4FAC081bf1f309aDC325306
        //ABI


    }

    function getConversionRate() public {}
    function getVerion() public view returns (uint256){

       return AggregatorV3Interface(0x694AA1769357215DE4FAC081bf1f309aDC325306).version();
    }                               
}

Above is the code and below is the error i am facing -

creation of FundMe pending...
creation of FundMe errored: [TIMEOUT] Timeout for call deployMetadataOf from udapp

I also see a warning when I deploy the contract, I have attached the image.
![Screenshot 2024-10-14 083957](https://github.com/user-attachments/assets/2d07ef68-7efd-407b-9d96-d70e068f15fc)

**Reply by PranavM191** [reply]
After I send transaction I face the following error and the contract is not deployed
![JSON error after deployment](https://github.com/user-attachments/assets/f5954bf8-9de0-4c33-94b1-1f277278b257)

**Reply by EngrPips** [reply]
Hello @PranavM191, Please use `wallet connect` option to connect Remix to your metamask and things should deploy correctly.

**Reply by KaneK31** [reply]
Also having the same issue, I am just using the remixVM for now and when I call the getPrice function it returns nothing

**Reply by EngrPips** [reply]
If you are using the price feed address for the sepolia test net on `remixVM` it won't work. You should deploy to the sepolia chain if you want to get price using the sepolia chain price feed.
